implementation/src/main/java/io/smallrye/config/ProfileConfigSourceInterceptor.java

```diff
@@ -27,10 +30,11 @@
         }
     };
 
-    private final String profile;
+    private final String[] profiles;
```

Comment:
I think it is better to be in a list. In this way, if someone specifies conflicting config profiles such as test, prod. With the list, you can specify the order say the first one takes precedence.

Reply:
With the array as well. In fact, priority is set with the first profile in the list and profile values are read in order. If one is found, that is the one returned.

Reply:
They are all ordered for sure. I just thought List is obvious. I think it is best to document which value is to be used.